### PR TITLE
Add course deletion request into dossier details page

### LINKED ISSUE #175: Finalize frontend to create a course deletion request
Implement UI where the client can request a course deletion in the dossier details page.

[Story Points: 2]
[Priority: High]
[Risk: Medium]

This is a subtask of https://github.com/JamesPartsafas/ConcordiaCurriculumManager/issues/135

### PR DESCRIPTION
**Pull request by kennyphan100**
This PR closes #175

### CODE REVIEW

**Review by Dinferno-lol — COMMENTED** [review]
Seems fine but I'd have a second opinion review it before merging.

**Comment by JamesPartsafas** [comment]
Looks good, but since everything is linked up now, can you also fix the delete-course page not working? If you try to create a deletion request, it always fails because there's no ResourceImplication field that gets sent.

**Comment by kennyphan100** [comment]
> Looks good, but since everything is linked up now, can you also fix the delete-course page not working? If you try to create a deletion request, it always fails because there's no ResourceImplication field that gets sent.

I fixed up the fields ResourceImplication and Rationale. The delete-course page works fine on my end. I am able to create a request.

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**
